drivers/axi_core/axi_dmac/axi_dmac.c

```diff
@@ -104,6 +154,33 @@ int32_t axi_dmac_transfer(struct axi_dmac *dmac,
 	default:
 		return FAILURE; // Other directions are not supported yet
 	}
+
+	if ((size - 1) > dmac->transfer_max_size) {
```

Comment:
why size - 1 and not size?

Reply:
Can this implementation be added to the current one? I mean if the transfer is less than transfer_max_size to fit in this logic too.

Reply:
The register will always do +1 to the value in the register size. the 'transfer_max_size' variable represents the maximum value you can write in the register.